src/test/binaryutils_test.cpp

```diff
@@ -597,4 +596,41 @@ INSTANTIATE_TEST_SUITE_P(
     [](const ::testing::TestParamInfo<BinaryUtilsTFSInvalidPrecisionTest::ParamType>& info) {
         return toString(info.param);
     });
+
+class BinaryUtilsTestKFSRawInputsContents : public ::testing::Test {
+public:
+    ::KFSRequest::InferInputTensor requestTensor;
+    std::string buffer;
+    void SetUp() override {
+        requestTensor.mutable_shape()->Add(1);
+        requestTensor.set_datatype("BYTES");
+
+        size_t filesize;
+        std::unique_ptr<char[]> image_bytes;
+
+        readRgbJpg(filesize, image_bytes);
+        buffer.append(image_bytes.get(), filesize);
+    }
+};
+
+TEST_F(BinaryUtilsTestKFSRawInputsContents, Positive) {
+    uint8_t rgb_expected_tensor[] = {0x24, 0x1b, 0xed};
+
+    ov::Tensor tensor;
+
+    std::shared_ptr<TensorInfo> tensorInfo = std::make_shared<TensorInfo>("", ovms::Precision::U8, ovms::Shape{1, 1, 1, 3}, Layout{"NHWC"});
+
+    ASSERT_EQ(convertBinaryRequestTensorToOVTensor(this->requestTensor, tensor, tensorInfo, &this->buffer), ovms::StatusCode::OK);
+    ASSERT_EQ(tensor.get_size(), 3);
+    uint8_t* ptr = static_cast<uint8_t*>(tensor.data());
+    EXPECT_EQ(std::equal(ptr, ptr + tensor.get_size(), rgb_expected_tensor), true);
+}
+
+TEST_F(BinaryUtilsTestKFSRawInputsContents, Negative_batchSizeBiggerThan1) {
+    ov::Tensor tensor;
+
+    std::shared_ptr<TensorInfo> tensorInfo = std::make_shared<TensorInfo>("", ovms::Precision::U8, ovms::Shape{2, 1, 1, 3}, Layout{"NHWC"});
+
+    ASSERT_EQ(convertBinaryRequestTensorToOVTensor(this->requestTensor, tensor, tensorInfo, &this->buffer), ovms::StatusCode::INVALID_BATCH_SIZE);
```

Comment:
Is this single test enough to cover all negative cases with last parameter set to non nullptr?